### PR TITLE
fix: Update email validation

### LINKED ISSUE #4399: [Enhancement] Add specific error in Registration validation
**Is your feature request related to a problem? Please describe.**

To reproduce the error:
1. Type any random email without any '@' or '.com/.in etc'
2. Password should be of 8 characters.
3. When you submit, it will show 'Unexpected Error Occurred'

But doesn't show which type of error or which important thing I missed in which specific input. 

Screenshot (466)

**Describe the solution you'd like**

I suggest that after any validation error, Error should be focussed on that input and error message  reason should be specific.

### PR DESCRIPTION
**Pull request by himanshuvarandani**
Fixes #4399 

#### Short description of what this resolves:
The email specified in this issue is invalid, so this pull request resolves the error of validating an email without ".com", ".io", etc.
The code for validating an email id using the "email" identifier which is validating the email id without ".com", so by changing the email identifier by regular expression resolves this issue.

#### Changes proposed in this pull request:
Changes "email" identifier by "regExp[/^[A-Za-z0-9_]([\.-]?[A-Za-z0-9_])*@[A-Za-z0-9_]([\.-]?[A-Za-z0-9_])*(\.[A-Za-z0-9_]{2,3})+$/]" in app/components/forms/login-form.js 
-
-
-

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia).
- [x] My branch is up-to-date with the Upstream `development` branch.
- [ ] The acceptance, integration, unit tests and linter pass locally with my changes 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by iamareebjamal** [comment]
It seems that those emails are invalid but they are not. We won't be adding custom regex to validate emails. Instead, the app should handle errors from backend gracefully and show proper error

File: app/controllers/register.js [hunk 1]
```diff
@@ -31,6 +31,8 @@ export default class RegisterController extends Controller {
       .catch(reason => {
         if (reason && Object.prototype.hasOwnProperty.call(reason, 'errors') && reason.errors[0].status === 409) {
           this.set('errorMessage', this.l10n.t('User already exists.'));
+        } else if (reason && Object.prototype.hasOwnProperty.call(reason, 'errors') && reason.errors[0].status === '422') {
```
Comment (iamareebjamal) [inline app/controllers/register.js:34]:
```suggestion
        } else if (reason?.errors[0]?.status === '422') {
```

**Comment by himanshuvarandani** [comment]
@iamareebjamal please review this pr again.
What I have to do next in this pr to merge?

**Comment by iamareebjamal** [comment]
Will test it today and merge. Thanks for reminding

**Review by iamareebjamal — APPROVED** [review]

**Pull request merged**